### PR TITLE
[xcode12] Merge xcode12 into d16-8

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Comment by mandel-macaque** [comment]
build

**Review by rolfbjarne — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Comment by tj-devel709** [comment]
So we are mostly looking to make sure the merge conflicts were settled the correct way right? We're not looking to fix whitespace and things like that?

**Comment by mandel-macaque** [comment]
@tj-devel709 just merge, we do not fix anything in this branch, except possible issues like the one that Rolf fixed in the last commit. This is just brings all the commits from xcode12.

**Review by tj-devel709 — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by whitneyschmidt — APPROVED** [review]

File: tests/cecil-tests/TaskAssemblyTests.cs [hunk 1]
```diff
@@ -40,6 +40,8 @@ bool IsTaskType (TypeDefinition? td)
 		[Test]
 		public void EnsureOnlyCodeInBaseTasks ()
 		{
+			if (assembly.Contains ("Xamarin.Mac.Tasks.dll"))
```
Comment (rolfbjarne) [inline tests/cecil-tests/TaskAssemblyTests.cs:43]: This should check `Configuration.include_mac` and Assert.Ignore if true. That way we don't have to remember to turn it back on later.

File: tests/dotnet/UnitTests/ProjectTest.cs [hunk 2]
```diff
@@ -83,6 +84,8 @@ public void BuildMyWatchApp ()
 		[TestCase ("macOS")]
 		public void BuildMyClassLibrary (string platform)
 		{
+			if (platform == "macOS")
```
Comment (rolfbjarne) [inline tests/dotnet/UnitTests/ProjectTest.cs:87]: Same: this should check `Configuration.include_mac` and Assert.Ignore if true.

File: tests/dotnet/UnitTests/ProjectTest.cs [hunk 3]
```diff
@@ -49,6 +49,7 @@ public void BuildMySingleView ()
 		[Test]
 		public void BuildMyCocoaApp ()
 		{
+			Assert.Ignore ("Ignore until Xamarin.Mac is re-enabled. Issue: https://github.com/xamarin/xamarin-macios/issues/9680");
```
Comment (rolfbjarne) [inline tests/dotnet/UnitTests/ProjectTest.cs:52]: Same: this should check `Configuration.include_mac` and Assert.Ignore if true.

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Review by spouliot — APPROVED** [review]

**Comment by mandel-macaque** [comment]
Last failure is a known network related issue: https://github.com/xamarin/maccore/issues/2103

**Pull request merged**